scripts/simtest/codecov.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash -e
+# Copyright (c) Mysten Labs, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+# verify that git repo is clean
+if [[ -n $(git status -s) ]]; then
```

Comment:
@mystenmark, why do we need this? This workflow is going to be run on `ubuntu-ghcloud` GH runner. So there should never be the case when repo is not clean

Reply:
someone could run the script locally (e.g. to test it) and it could clobber local files

Reply:
@mystenmark 
```
  ./scripts/simtest/codecov.sh
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_INCREMENTAL: 0
    CARGO_NET_RETRY: 10
    RUSTUP_MAX_RETRIES: 10
    RUST_BACKTRACE: short
    RUSTDOCFLAGS: -D warnings
    CACHE_ON_FAILURE: false
Working directory is not clean. Please commit all changes before running this script.  ./scripts/simtest/codecov.sh
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_INCREMENTAL: 0
    CARGO_NET_RETRY: 10
    RUSTUP_MAX_RETRIES: 10
    RUST_BACKTRACE: short
    RUSTDOCFLAGS: -D warnings
    CACHE_ON_FAILURE: false
Working directory is not clean. Please commit all changes before running this script.
```
Is what I was afraid of when this script gets run. I feel like these are two different cases here